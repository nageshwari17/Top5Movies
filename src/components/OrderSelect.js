import React from 'react';

const OrderSelect = (props) => {
  const { select, setOrder } = props;
  return (
    <div className="row">
      <div className="col s12 order-select">
        <div className="col s12 m6 l4">
          <select onChange={(e) => { console.log(e.target.value); setOrder(e.target.value); } } data-testid="sortBox">
            <option value="">Sort By</option>
            {select &&
              select.items.map((item) => {
                return (
                  <option value={item.valueToOrderBy} key={item.valueToOrderBy}>
                    {item.label}
                  </option>
                );
              })}
          </select>
        </div>
      </div>
    </div>
  );
};

export default OrderSelect;
