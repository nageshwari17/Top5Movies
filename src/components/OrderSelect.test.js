import React from 'react';
import { configure, shallow } from 'enzyme';
import OrderSelect from './OrderSelect';
import Adapter from 'enzyme-adapter-react-16';
import sinon, { expectation } from 'sinon';
import data from '../db.json';
const [ select, list ] = data.components;

configure({ adapter: new Adapter() });

// test('render Markdown in preview mode', () => {
//   const spy = sinon.spy();
//   const wrapper = shallow(<OrderSelect />);
//   console.log(wrapper.debug());
//   console.log(wrapper.instance());
//   jest.spyOn(wrapper.find('select'), 'setOrder');
//   wrapper.instance().setOrder('releaseDate');
//   expect(wrapper.instance().setOrder).toHaveBeenCalledTimes(1);
//   //wrapper.find('select').simulate('change', {},  { target: { value: 'dropDownOption' } });
//   //expect(spy).toHaveProperty('callCount', 1);
// });

test('check for submit functionality', () => { 
  const wrapper = shallow(<OrderSelect />); 
  const selectdOrder = (val) => {
    dispatch(sortlist('releaseDate'));
    setQuickLink(0);
  };
  /*wrapper.setState({ 
    isFormSubmitted: false, 
    checkBoxChecked: true
  });  */
  wrapper.setProps({  
    setOrder: {
      selectdOrder
    } 
  }); 
  wrapper.find('select').simulate('change'); 
  expect(wrapper.find('select').length).toEqual(1);
  //expect(wrapper.instance().state.isFormSubmitted).toEqual(true);         
});
