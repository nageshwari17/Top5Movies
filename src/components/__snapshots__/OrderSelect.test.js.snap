// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Markdown in preview mode 1`] = `ShallowWrapper {}`;
